src/timeseriesflattener/feature_spec_objects.py

```diff
@@ -326,38 +578,59 @@ def is_dichotomous(self) -> bool:
 
 
 class MinGroupSpec(BaseModel):
```

Comment:
```suggestion
class _MinGroupSpec(BaseModel):
```

Perhaps we should make this, `MinSpec` and `AnySpec` private?